src/inc/til/string.h

```diff
@@ -348,70 +241,244 @@ namespace til // Terminal Implementation Library. Also: "Today I Learned"
+    constexpr std::basic_string_view<T, Traits> trim(const std::basic_string_view<T, Traits>& str, const T ch) noexcept
     {
-        using view_type = std::basic_string_view<T, Traits>;
-
-        const auto needleLen = needle.size();
-        const auto idx = needleLen == 0 ? str.size() : str.find(needle);
-        const auto prefixIdx = std::min(str.size(), idx);
-        const auto suffixIdx = std::min(str.size(), prefixIdx + needle.size());
+        auto beg = str.data();
+        auto end = beg + str.size();
 
-        const view_type result{ str.data(), prefixIdx };
-#pragma warning(suppress : 26481) // Don't use pointer arithmetic. Use span instead
-        str = { str.data() + suffixIdx, str.size() - suffixIdx };
-        return result;
-    }
+        for (; beg != end && *beg == ch; ++beg)
+        {
+        }
 
-    constexpr std::string_view prefix_split(std::string_view& str, const std::string_view& needle) noexcept
-    {
-        return prefix_split<>(str, needle);
-    }
+        for (; beg != end && end[-1] == ch; --end)
+        {
+        }
 
-    constexpr std::wstring_view prefix_split(std::wstring_view& str, const std::wstring_view& needle) noexcept
-    {
-        return prefix_split<>(str, needle);
+        return { beg, end };
     }
 
-    // Give the arguments ("foo bar baz", " "), this method will
-    // * modify the first argument to "bar baz"
-    // * return "foo"
-    // If the needle cannot be found the "str" argument is returned as is.
     template<typename T, typename Traits>
-    constexpr std::basic_string_view<T, Traits> prefix_split(std::basic_string_view<T, Traits>& str, T ch) noexcept
+    struct split_iterator
     {
-        using view_type = std::basic_string_view<T, Traits>;
+        struct sentinel
+        {
+        };
+
+        struct iterator
+        {
+            using iterator_category = std::forward_iterator_tag;
+            using value_type = std::basic_string_view<T, Traits>;
+            using reference = value_type&;
+            using pointer = value_type*;
+            using difference_type = std::ptrdiff_t;
+
+            explicit constexpr iterator(split_iterator& p) noexcept :
+                _iter{ p }
+            {
+            }
 
-        const auto idx = str.find(ch);
-        const auto prefixIdx = std::min(str.size(), idx);
-        const auto suffixIdx = std::min(str.size(), prefixIdx + 1);
+            value_type operator*() const noexcept
+            {
+#ifndef NDEBUG
+                // Don't call operator*() twice per iteration.
+                // Since C++ has inflexible iterators we can't implement a split iterator efficiently:
+                // We essentially need a do-while loop but iterators only offer for() loop semantics.
+                // This forces us to either seek to the first token in the constructor and then again on

```

Comment:
qq: why prefer the one that makes it unsafe/hazardous to handle, instead of the one that makes it safe?